# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./configure.sh --prefix=

# build
make

# install
install -d $DESTDIR/bin
install a.out $DESTDIR/bin/m4

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/m4-build.pc << EOF
Name: m4
Version: 1.4.18
Description: GNU M4
Requires: $REQUIRES

devel=\\
/bin/m4
EOF

exit

# patch
--- m4-1.4.18.orig/configure.sh
+++ m4-1.4.18/configure.sh
@@ -0,0 +1,188 @@
+#!/bin/sh
+# Shell script to configure GNU M4 without relying on `autotools'.
+
+# Copyright (C) 1991, 1993-1994, 2004-2014, 2016 Free Software
+# Foundation, Inc.
+#
+# This file is part of GNU M4.
+#
+# GNU M4 is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# GNU M4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+CC=${CC-gcc}
+CFLAGS=${CFLAGS-"-g -O2"}
+SED=${SED-sed}
+
+prefix=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) prefix=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+srcdir=`dirname $0`
+
+VERSION=`cat $srcdir/.version`
+
+test -d lib || mkdir lib
+
+cat > lib/configmake.h << EOF
+/* lib/configmake.h.  Generated from configure.sh.  */
+
+#define PREFIX "$prefix"
+#define BINDIR "$prefix/bin"
+#define LIBDIR "$prefix/lib"
+EOF
+
+echo 'int main() { return __freadahead(); }' | $CC -xc - 2>/dev/null && HAVE___FREADAHEAD=1 || { HAVE___FREADAHEAD=0; FREADAHEAD_OBJ=\$\(srcdir\)/lib/freadahead.o; }
+
+cat > lib/config.h << EOF
+/* lib/config.h.  Generated from configure.sh.  */
+
+#define _FORTIFY_SOURCE 2
+#define GNULIB_CLOSE_STREAM 1
+#define GNULIB_DIRNAME 1
+#define GNULIB_FD_SAFER_FLAG 1
+#define GNULIB_FILENAMECAT 1
+#define GNULIB_FOPEN_SAFER 1
+#define GNULIB_PIPE2_SAFER 1
+#define HAVE___FREADAHEAD $HAVE___FREADAHEAD
+#define HAVE_ALLOCA 1
+#define HAVE_DECL_FERROR_UNLOCKED 1
+#define HAVE_DECL_FPUTS_UNLOCKED 1
+#define HAVE_DECL_GETC_UNLOCKED 1
+#define HAVE_DECL_PROGRAM_INVOCATION_NAME 1
+#define HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME 1
+#define HAVE_DECL_SIGALTSTACK 1
+#define HAVE_DECL_STRERROR_R 1
+#define HAVE_LANGINFO_CODESET 1
+#define HAVE_LONG_LONG_INT 1
+#define HAVE_MEMPCPY 1
+#define HAVE_MINMAX_IN_SYS_PARAM_H 1
+#define HAVE_PIPE 1
+#define HAVE_SIGALTSTACK 1
+#define HAVE_STACK_OVERFLOW_HANDLING 1
+#define HAVE_STACK_T 1
+#define HAVE_STDIO_EXT_H 1
+#define HAVE_WORKING_O_NOFOLLOW 1
+#define PACKAGE "m4"
+#define PACKAGE_BUGREPORT "bug-m4@gnu.org"
+#define PACKAGE_NAME "GNU M4"
+#define PACKAGE_STRING "GNU M4 $VERSION"
+#define PACKAGE_URL "http://www.gnu.org/software/m4/"
+#define RENAME_OPEN_FILE_WORKS 1
+#define SIGNAL_SAFE_LIST 1
+#define SYSCMD_SHELL "/bin/sh"
+#define USE_UNLOCKED_IO 1
+#define VERSION "$VERSION"
+#define _GL_INLINE extern inline __attribute__ ((__gnu_inline__))
+#ifdef __GNUC_STDC_INLINE__
+# define _GL_EXTERN_INLINE extern inline
+#else
+# define _GL_EXTERN_INLINE extern
+#endif
+#define _GL_INLINE_HEADER_BEGIN \\
+    _Pragma ("GCC diagnostic push") \\
+    _Pragma ("GCC diagnostic ignored \"-Wmissing-prototypes\"") \\
+    _Pragma ("GCC diagnostic ignored \"-Wmissing-declarations\"") \\
+    _Pragma ("GCC diagnostic ignored \"-Wsuggest-attribute=const\"")
+#define _GL_INLINE_HEADER_END \\
+    _Pragma ("GCC diagnostic pop")
+#define restrict __restrict
+#define _GL_ATTRIBUTE_PURE __attribute__ ((__pure__))
+EOF
+
+cat > Makefile << EOF
+# Makefile.  Generated from configure.sh.
+
+VERSION = $VERSION
+
+CC = $CC
+CFLAGS = $CFLAGS
+SED = $SED
+
+srcdir = $srcdir
+
+CFLAGS += -Ilib -I\$(srcdir)/lib -D_GNU_SOURCE
+
+FREADAHEAD_OBJ = $FREADAHEAD_OBJ
+
+OBJECTS = \\
+	\$(srcdir)/lib/basename-lgpl.o \\
+	\$(srcdir)/lib/c-stack.o \\
+	\$(srcdir)/lib/clean-temp.o \\
+	\$(srcdir)/lib/cloexec.o \\
+	\$(srcdir)/lib/close-stream.o \\
+	\$(srcdir)/lib/closein.o \\
+	\$(srcdir)/lib/closeout.o \\
+	\$(srcdir)/lib/dup-safer.o \\
+	\$(srcdir)/lib/error.o \\
+	\$(srcdir)/lib/execute.o \\
+	\$(srcdir)/lib/exitfail.o \\
+	\$(srcdir)/lib/fatal-signal.o \\
+	\$(srcdir)/lib/fd-safer.o \\
+	\$(srcdir)/lib/filenamecat.o \\
+	\$(srcdir)/lib/filenamecat-lgpl.o \\
+	\$(srcdir)/lib/fopen-safer.o \\
+	$FREADAHEAD_OBJ \\
+	\$(srcdir)/lib/getprogname.o \\
+	\$(srcdir)/lib/gl_avltree_oset.o \\
+	\$(srcdir)/lib/gl_linkedhash_list.o \\
+	\$(srcdir)/lib/localcharset.o \\
+	\$(srcdir)/lib/malloca.o \\
+	\$(srcdir)/lib/memchr2.o \\
+	\$(srcdir)/lib/mkstemp-safer.o \\
+	\$(srcdir)/lib/obstack.o \\
+	\$(srcdir)/lib/pipe-safer.o \\
+	\$(srcdir)/lib/progname.o \\
+	\$(srcdir)/lib/quotearg.o \\
+	\$(srcdir)/lib/regex.o \\
+	\$(srcdir)/lib/spawn-pipe.o \\
+	\$(srcdir)/lib/tmpdir.o \\
+	\$(srcdir)/lib/verror.o \\
+	\$(srcdir)/lib/version-etc.o \\
+	\$(srcdir)/lib/version-etc-fsf.o \\
+	\$(srcdir)/lib/wait-process.o \\
+	\$(srcdir)/lib/xalloc-die.o \\
+	\$(srcdir)/lib/xasprintf.o \\
+	\$(srcdir)/lib/xmalloc.o \\
+	\$(srcdir)/lib/xmalloca.o \\
+	\$(srcdir)/lib/xprintf.o \\
+	\$(srcdir)/lib/xvasprintf.o \\
+	\$(srcdir)/src/m4.o \\
+	\$(srcdir)/src/builtin.o \\
+	\$(srcdir)/src/debug.o \\
+	\$(srcdir)/src/eval.o \\
+	\$(srcdir)/src/format.o \\
+	\$(srcdir)/src/freeze.o \\
+	\$(srcdir)/src/input.o \\
+	\$(srcdir)/src/macro.o \\
+	\$(srcdir)/src/output.o \\
+	\$(srcdir)/src/path.o \\
+	\$(srcdir)/src/symtab.o
+
+a.out: \$(OBJECTS)
+	@echo linking GNU M4 \$(VERSION) \(\`\$(SED) -n '1s/^\([^ ]*\).*/\1/p' \$(srcdir)/ChangeLog\`\)...
+	\$(CC) $^
+
+all: a.out
+
+install: all
+	install -D a.out \$(DESTDIR)$prefix/bin/m4
+
+clean:
+	\$(RM) a.out \$(OBJECTS)
+EOF
--- m4-1.4.18.orig/lib/binary-io.h
+++ m4-1.4.18/lib/binary-io.h
@@ -20,6 +20,9 @@
 /* For systems that distinguish between text and binary I/O.
    O_BINARY is guaranteed by the gnulib <fcntl.h>. */
 #include <fcntl.h>
+#ifndef O_BINARY
+# define O_BINARY 0
+#endif
 
 /* The MSVC7 <stdio.h> doesn't like to be included after '#define fileno ...',
    so we include it here first.  */
--- m4-1.4.18.orig/lib/error.c
+++ m4-1.4.18/lib/error.c
@@ -44,6 +44,12 @@
 # define _GL_ARG_NONNULL(a)
 #else
 # include "getprogname.h"
+#ifndef _GL_ATTRIBUTE_FORMAT_PRINTF
+# define _GL_ATTRIBUTE_FORMAT_PRINTF(a, b) __attribute__ ((format(printf, a, b)))
+#endif
+#ifndef _GL_ARG_NONNULL
+# define _GL_ARG_NONNULL(a) __attribute__ ((nonnull a))
+#endif
 #endif
 
 #if USE_UNLOCKED_IO
--- m4-1.4.18.orig/lib/unlocked-io.h
+++ m4-1.4.18/lib/unlocked-io.h
@@ -86,14 +86,18 @@
 #  undef fread
 #  define fread(w,x,y,z) fread_unlocked (w,x,y,z)
 # else
+#  ifndef fread_unlocked
 #  define fread_unlocked(w,x,y,z) fread (w,x,y,z)
+#  endif
 # endif
 
 # if HAVE_DECL_FWRITE_UNLOCKED
 #  undef fwrite
 #  define fwrite(w,x,y,z) fwrite_unlocked (w,x,y,z)
 # else
+#  ifndef fwrite_unlocked
 #  define fwrite_unlocked(w,x,y,z) fwrite (w,x,y,z)
+#  endif
 # endif
 
 # if HAVE_DECL_GETC_UNLOCKED

# source
https://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz
